Components/Motors/VESC.cs

```diff
@@ -9,31 +9,97 @@ namespace Scarlet.Components.Motors
+            CAN_PACKET_STATUS = 9,
+        }
+        #endregion
+
+        private readonly sbyte MOTOR_MAX_RPM;
+        private readonly int ERPM_PER_RPM;
+
+        private IFilter<sbyte> RPMFilter; // Filter for speed output
         private readonly IUARTBus UARTBus;
+        private readonly ICANBus CANBus;
         private readonly int CANForwardID;
-        private readonly float MaxSpeed;
+        private readonly uint CANID;
+        private readonly sbyte MaxRPM;
+        private readonly bool IsCAN;
 
         private bool OngoingSpeedThread; // Whether or not a thread is running to set the speed
         private bool Stopped; // Whether or not the motor is stopped
         public float TargetSpeed { get; private set; } // Target speed (-1.0 to 1.0) of the motor
+        public sbyte TargetRPM { get; private set; } // Target RPM of the motor
 
         /// <summary> Initializes a VESC Motor controller </summary>
         /// <param name="UARTBus"> UART output to control the motor controller </param>
         /// <param name="MaxSpeed"> Limiting factor for speed (should never exceed + or - this val) </param>
         /// <param name="CANForwardID"> CAN ID of the motor controller (-1 to disable CAN forwarding) </param>
-        /// <param name="SpeedFilter"> Filter to use with MC. Good for ramp-up protection and other applications </param>
-        public VESC(IUARTBus UARTBus, float MaxSpeed, int CANForwardId = -1, IFilter<float> SpeedFilter = null)
+        /// <param name="RPMFilter"> Filter to use with MC. Good for ramp-up protection and other applications </param>
+        public VESC(IUARTBus UARTBus, float MaxSpeed, int CANForwardID = -1, IFilter<sbyte> RPMFilter = null, int ErpmPerRpm = 518, sbyte MotorMaxRpm = 60)
+            : this(UARTBus, (sbyte)(MaxSpeed * MotorMaxRpm), CANForwardID, RPMFilter) { }
+
+        public VESC(IUARTBus UARTBus, sbyte MaxRPM, int CANForwardID = -1, IFilter<sbyte> RPMFilter = null, int ErpmPerRpm = 518, sbyte MotorMaxRpm = 60)
         {
+            IsCAN = false;
             this.UARTBus = UARTBus;
             this.UARTBus.BaudRate = UARTRate.BAUD_115200;
             this.UARTBus.BitLength = UARTBitCount.BITS_8;
             this.UARTBus.StopBits = UARTStopBits.STOPBITS_1;
             this.UARTBus.Parity = UARTParity.PARITY_NONE;
-            this.CANForwardID = CANForwardId;
-            this.MaxSpeed = Math.Abs(MaxSpeed);
-            this.Filter = SpeedFilter;
-            this.SetSpeedDirectly(0.0f);
+            this.CANForwardID = CANForwardID;
+            this.MaxRPM = Math.Abs(MaxRPM);
+            this.ERPM_PER_RPM = Math.Abs(ErpmPerRpm);
+            this.MOTOR_MAX_RPM = Math.Abs(MotorMaxRpm);
+            this.RPMFilter = RPMFilter;
+            this.SetRPMDirectly(0);
+            SetSpeedThreadFactory().Start();
+        }
+
+        public VESC(ICANBus CANBus, float MaxSpeed, uint CANID, IFilter<sbyte> RPMFilter = null, int ErpmPerRpm = 518, sbyte MotorMaxRpm = 60)
```

Comment:
I really like how this is one class for both communications types, I was hoping you'd do this. Nice :)